### ISSUE TITLE
Deleting an item from a dynamic form does not reflect correctly on the onValuesChange function

### ISSUE THREAD

**Issue by budupaul** [ant-design/ant-design#27576]
- [x] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[![Edit on CodeSandbox](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/dynamic-form-nest-items-antd480-forked-ncimn?file=/index.js)

### Steps to reproduce
1.Add 2 rows to the dynamic form and enter values for them.
2. Delete the last one.
3. Check values in the onValuesChange function.

### What is expected?
The values should now be an array with only one element, the first one.

### What is actually happening?
The values are actually an array with 2 elements with the properties of the last one having been changed to undefined.

| Environment | Info |
|---|---|
| antd | undefined |
| React | 16.14 |
| System | Windows 10 Pro |
| Browser | Chrome |

---
Basically, instead of deleting the item, it just sets its attributes to undefined.

**Comment by budupaul** [comment]
@zombieJ What's the antd version from which this issue is resolved?

**Comment by Jdash99** [comment]
the bugs persists in the last present version 4.8.4

**Comment by zombieJ** [comment]
https://codesandbox.io/s/dynamic-form-nest-items-antd480-forked-n5l7r

Tested. Seems it work as expect.

**Comment by Jdash99** [comment]
Tested again in my code and it is working as expected. Thank you.
